soroban-env-host/src/budget/limits.rs

```diff
@@ -96,7 +96,9 @@ impl DepthLimiter for BudgetImpl {
     // `leave` should be called in tandem with `enter` such that the depth
     // doesn't exceed the initial depth limit.
     fn leave(&mut self) -> Result<(), HostError> {
-        self.depth_limit = self.depth_limit.saturating_add(1);
+        self.depth_limit = self.depth_limit.checked_add(1).ok_or_else(|| {
+            Error::from_type_and_code(ScErrorType::Value, ScErrorCode::ArithDomain)
```

Comment:
This looks like an internal error to me - this could only ever occur if we call `leave` without `enter`.